source/slang/slang.cpp

```diff
@@ -91,6 +97,55 @@ Session::Session()
     addBuiltinSource(hlslLanguageScope, "hlsl", getHLSLLibraryCode());
 }
 
+ISlangUnknown* Session::getInterface(const Guid& guid)
+{
+    if(guid == IID_ISlangUnknown || guid == IID_IGlobalSession)
+        return asExternal(this);
+    return nullptr;
+}
+
+SLANG_NO_THROW SlangResult SLANG_MCALL Session::createSession(
+    slang::SessionDesc const&  desc,
+    slang::ISession**          outSession)
+{
+    auto linkage = new Linkage(this);
```

Comment:
Might have been better to use a smart pointer here, and then detach at the very end. Means returns/throw inbetween don't leak.

Reply:
Oh yeah, that looks like an oversight.